### PR TITLE
Intractive segmentation

### PR DESCRIPTION
**Pull request by faezehamjadi**

### CODE REVIEW

File: deep-learning/_posts/2018-06-27-Intractivesegmentation.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+### Description
+
+What is interactive segmentation?
+
+It allows the user to segment the desired region in the image by successively placing positive
+(‘foreground’) and negative (‘background’) clicks. Every time a click is placed, the segmentation is updated. Once the segmentation matches the region intended by the user, the process concludes.
+
+In this paper, they trained a model which couples two convolution network. This model takes an image and the clicks as input and produces the segmentation on click object.
+
+First, a network synthesizes M segmentations f(x) = y, which y is a collection of segmentation masks. Second, a network g selects one of the M solutions synthesized by f. This intermediate representation y allows the network f to produce multiple clean segmentations and the network f produces
```
Comment (jodoin) [inline deep-learning/_posts/2018-06-27-Intractivesegmentation.md]: This intermediate representation y allows the network f to....

Split that sentence in 2

File: deep-learning/_posts/2018-06-27-Intractivesegmentation.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+plausible segments for another network(g) to select from. So, g takes a representation of the
+input and the synthesized segmentation masks, and produces a probability distribution over {1, . . . , M}.
+
+<img src="/deep-learning/images/IIS/1.png" width="1000">
+
+The input to the segmentation network f consists of the image X, clicks Sp(positive) and Sn(negative), distance transform where uses a Euclidean distance transformation and calculates the minimum Euclidean distance between a click point pixel and other pixels. In addition, they applied a VGG-19 network
+pre-trained on the ImageNet dataset to the
```
Comment (jodoin) [inline deep-learning/_posts/2018-06-27-Intractivesegmentation.md]: on the ImageNet dataset to => on ImageNet to

File: deep-learning/_posts/2018-06-27-Intractivesegmentation.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+image x and extract the feature maps. 
+
+To produce a diversity segmentations, they applied Jaccard distance to measure the distance between the ground truth segmentation mask A(p) and predicted mask B(p)at pixel p. In addition, the input clicks are soft constraints. where the Hadamard elementwise product is applied to Sp and predicted masks.
```
Comment (jodoin) [inline deep-learning/_posts/2018-06-27-Intractivesegmentation.md]: a diversity of segmentation,

soft constraints. where => soft constraints where

is applied between Sp and ...

File: deep-learning/_posts/2018-06-27-Intractivesegmentation.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+Finally, The selection network g takes as input and is trained
```
Comment (jodoin) [inline deep-learning/_posts/2018-06-27-Intractivesegmentation.md]: takes as input what?

File: deep-learning/_posts/2018-06-27-Intractivesegmentation.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+---
+
+### Description
+
+What is interactive segmentation?
+
+It allows the user to segment the desired region in the image by successively placing positive
+(‘foreground’) and negative (‘background’) clicks. Every time a click is placed, the segmentation is updated. Once the segmentation matches the region intended by the user, the process concludes.
+
+In this paper, they trained a model which couples two convolution network. This model takes an image and the clicks as input and produces the segmentation on click object.
```
Comment (lemairecarl) [inline deep-learning/_posts/2018-06-27-Intractivesegmentation.md]: on the clicked object?

File: deep-learning/_posts/2018-06-27-Intractivesegmentation.md (hunk 3 above)
Comment (lemairecarl) [reply deep-learning/_posts/2018-06-27-Intractivesegmentation.md]: Is is not clear from that paragraph how diversity is achieved. Do they inject noise?

Comment (faezehamjadi) [reply deep-learning/_posts/2018-06-27-Intractivesegmentation.md]: no, they don't inject the noise. They try to have different segmentation on an input. For training then they shuffle the order of these segments for selection network

File: deep-learning/_posts/2018-06-27-Intractivesegmentation.md (hunk 5 above)
Comment (faezehamjadi) [reply deep-learning/_posts/2018-06-27-Intractivesegmentation.md]: The input is an image plus a click point( instead of a bounding box).

**Pull request merged**